packages/material-ui-styled-engine-sc/src/StylesProvider/StylesProvider.js

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+
+let injectFirstNode;
+
+export function StylesProvider({ injectFirst, children }) {
+  React.useEffect(() => {
+    if (injectFirst && !injectFirstNode) {
+      const head = document.head;
+      injectFirstNode = document.createElement('style');
+      injectFirstNode.setAttribute('data-styled', 'active');
+      injectFirstNode.setAttribute('data-styled-version', '5.2.1');
+      head.insertBefore(injectFirstNode, head.firstChild);
```

Comment:
I tried it at some point but it didn't work, but it may have been related to other changes. Let me try again, it should definitely be cleaner than this. Also this doesn't work at the moment, the node is created but the styles are being added to a different node.

Reply:
I found out why it wasn't working, but it is awkward. The node added has to be style tag with `data-styled='active'`. Adding just a comment doesn't work. Basically, now it works, but we have first empty style tag, and then the one from styled-components.. Will leave it like this until we find a better alternative..